src/Text/Layout/Table/Cell/Formatted.hs

```diff
@@ -40,6 +43,29 @@ formatted
     -> Formatted a
 formatted = Format
 
+-- | Map over the formatting directives of a formatted value.
+reformat :: (String -> String)  -- ^ Function to operate on prefix text directives.
+         -> (String -> String)  -- ^ Function to operate on suffix text directives.
+         -> Formatted a  -- ^ The formatted value to operate on.
+         -> Formatted a
+reformat preF sufF = cataFormatted Empty Concat Plain (\p x s -> Format (preF p) x (sufF s))
+
+-- | Process a formatted value to produce an arbitrary value.
+-- This is the catamorphism for 'Formatted'.
+cataFormatted :: b  -- ^ Value of 'Empty'.
+              -> ([b] -> b)  -- ^ Function for operating over 'Concat'.
+              -> (a -> b)  -- ^ Function for operating over 'Plain'.
+              -> (String -> b -> String -> b)  -- ^ Function for operating over 'Format'.
```

Comment:
* One issue I see here is that we do not export the constructors.
* General question:  Do you have any use case in mind where this would be required externally?  This was meant as a sink type and not something the user should operate on.

Reply:
Not exporting the constructors is one of the reasons this is most useful. It essentially allows you to pattern match without access to the constructors.

The use case I have in mind is in hledger. We have text rendering functions which are usually displayed in tables or text, but we also have an ncurses-based backend. The formatting there is done in a monad, so instead of instituting a whole new rendering function, it would be nice to write a function which allows translating `Formatted` into the Monad-based rendering system.